cmd/cli/root.go

```diff
@@ -45,43 +42,21 @@ func NewRootCmd() *cobra.Command {
 		"api":     configflags.ClientAPIFlags,
 		"logging": configflags.LogFlags,
 	}
+	cfg := config.New()
 	RootCmd := &cobra.Command{
 		Use:   os.Args[0],
 		Short: "Compute over data",
 		Long:  `Compute over data`,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			repoDir, err := config.Get[string]("repo")
-			if err != nil {
-				util.Fatal(cmd, fmt.Errorf("failed to read --repo value: %w", err), 1)
-			}
-			if repoDir == "" {
-				// this error indicates `defaultRepo` was unable to find a default location and the user
-				// didn't provide on.
-				util.Fatal(cmd, fmt.Errorf("bacalhau repo not set, please use BACALHAU_DIR or --repo"), 1)
-			}
-			if _, err := setup.SetupBacalhauRepo(repoDir); err != nil {
-				util.Fatal(cmd, fmt.Errorf("failed to initialize bacalhau repo at '%s': %w", repoDir, err), 1)
-			}
-
-			if err := configflags.BindFlags(cmd, rootFlags); err != nil {
+			if err := configflags.BindFlags(cmd, cfg, rootFlags); err != nil {
```

Comment:
Review: We no longer create a config when the bacalhau binary runs. We instead create or open a repo in the places it's needed.